pkg/crc/network/httpproxy/proxy.go

```diff
@@ -45,6 +46,9 @@ func readProxyCAData(proxyCAFile string) (string, error) {
 	if proxyCAFile == "" || !filepath.IsAbs(proxyCAFile) {
 		return "", nil
 	}
+	if !crcos.FileExists(proxyCAFile) {
+		return "", nil
```

Comment:
Is there a validation happening later on? Ideally, most crc commands would not fail when the proxy config is invalid. It's probably only `crc config`, `crc setup` and `crc start` which should report a hard failure when the proxy is invalid, the other ones are not dependent on the proxy settings.

Reply:
> Is there a validation happening later on? 

no, there's validation when running `config set`. and then here when proxy is being setup which is very early on in `root.go`.  (we send telemetry data from each command, and that uses the proxy settings)

> It's probably only crc config, crc setup and crc start which should report a hard failure when the proxy is invalid, the other ones are not dependent on the proxy settings.

maybe for these commands we can re-validate (add a "re-validation" helper in proxy package) the proxy settings and this will be hard failure if validation fails

Reply:
Thinking about this again, doing this selective hard failure for `start` and `setup` seems not to be necessary, as we have added validation of proxy config from environment, now it should not possible to configure invalid proxy settings at all via the environment or via `crc config set`

for the `proxy-ca-file` we have validation while `config set` is used to make sure the file exists, and with the conversion of relative to absolute paths, user shouldn't end up in a situation where they have invalid path to `proxy-ca-file`

now the only way invalid proxy settings can end up in the config is by manually editing the config file, which i think should be a hard failure for all the commands, WDYT?

Reply:
> for the proxy-ca-file we have validation while config set is used to make sure the file exists, and with the conversion of relative to absolute paths, user shouldn't end up in a situation where they have invalid path to proxy-ca-file

it's a user problem, but
```
$ crc config set proxy-ca-file foo.crt
$ rm foo.crt
```

> Thinking about this again, doing this selective hard failure for start and setup seems not to be necessary, as we have added validation of proxy config from environment, now it should not be possible to configure invalid proxy settings at all via the environment or via crc config set

We validate the proxy config, and if for example `HTTP_PROXY` is invalid in the environment, we ignore it right?
If the user made a typo when setting it, it's better to make this very visible than ignoring it, no?

Reply:
If there are already warnings for all of this every time crc starts up, it might be enough ,I'll have to check how it looks :)

Reply:
> We validate the proxy config, and if for example HTTP_PROXY is invalid in the environment, we ignore it right?
If the user made a typo when setting it, it's better to make this very visible than ignoring it, no?

yes i changed that its not silently ignoring it now. If the configured `proxy-ca-file` path is not present, it returns an error which is a hard failure for the all the commands, also with a msg asking user to either unset or correct the config